vllm_ascend/worker/model_runner_v1.py

```diff
@@ -687,6 +811,92 @@ def apply_grammar_bitmask(
         )
         return logits.to(self.device).to(logits_dtype)
 
+    def get_spec_token_ids(
+        self,
+        valid_sampled_token_ids: list[list[int]],
+        sampling_metadata: SamplingMetadata,
+        scheduler_output: "SchedulerOutput",
+        spec_decode_metadata: SpecDecodeMetadata,
+        positions: torch.Tensor,
+        num_scheduled_tokens: int,
+        hidden_states: torch.Tensor,
+        attn_metadata: SpecDecodeMetadata,
+    ) -> Optional[list[list[int]]]:
+        if not self.use_spec_decode:
+            # Speculative decoding is not enabled.
+            spec_token_ids = None
+        elif self.speculative_config.method == "ngram":
+            assert isinstance(self.drafter, NgramProposer)
+            spec_token_ids = self.generate_draft_token_ids(
+                valid_sampled_token_ids, sampling_metadata)
+        elif self.speculative_config.method == "eagle":
```

Comment:
Add a note/todo here to highlight that `eagle` mode doesn't work currently.
And if `eagle` doesn't work now, i think we'd better raise error here directly.  And complete the function in the future PR.

```
 elif self.speculative_config.method == "eagle":
    raise NotImplementedError("eagle method for spec decode doesn't work on vllm-ascend currently")
```